dataproc/src/main/java/Quickstart.java

```diff
@@ -0,0 +1,166 @@
+
+    // Configure the settings for the cluster controller client
+    ClusterControllerSettings clusterControllerSettings =
+        ClusterControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+    // Configure the settings for the job controller client
+    JobControllerSettings jobControllerSettings =
+        JobControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+    // Create both a cluster controller client and job controller client with the configured
+    // settings. The client only needs to be created once and can be reused for multiple requests.
+    // Using a try-with-resources closes the client, but this can also be done manually with
+    // the .close() method.
+    try (ClusterControllerClient clusterControllerClient =
+            ClusterControllerClient.create(clusterControllerSettings);
+        JobControllerClient jobControllerClient =
+            JobControllerClient.create(jobControllerSettings)) {
+      // Configure the settings for our cluster
+      InstanceGroupConfig masterConfig =
+          InstanceGroupConfig.newBuilder()
+              .setMachineTypeUri("n1-standard-1")
+              .setNumInstances(1)
+              .build();
+      InstanceGroupConfig workerConfig =
+          InstanceGroupConfig.newBuilder()
+              .setMachineTypeUri("n1-standard-1")
+              .setNumInstances(2)
+              .build();
+      ClusterConfig clusterConfig =
+          ClusterConfig.newBuilder()
+              .setMasterConfig(masterConfig)
+              .setWorkerConfig(workerConfig)
+              .build();
+      // Create the cluster object with the desired cluster config
+      Cluster cluster =
+          Cluster.newBuilder().setClusterName(clusterName).setConfig(clusterConfig).build();
+
+      // Create the Cloud Dataproc cluster
+      OperationFuture<Cluster, ClusterOperationMetadata> createClusterAsyncRequest =
+          clusterControllerClient.createClusterAsync(projectId, region, cluster);
+      Cluster response = createClusterAsyncRequest.get();
+      System.out.printf("Cluster created successfully: %s", response.getClusterName());
+
+      // Configure the settings for our job
+      JobPlacement jobPlacement = JobPlacement.newBuilder().setClusterName(clusterName).build();
+      PySparkJob pySparkJob = PySparkJob.newBuilder().setMainPythonFileUri(jobFilePath).build();
+      Job job = Job.newBuilder().setPlacement(jobPlacement).setPysparkJob(pySparkJob).build();
+
+      // Submit an asynchronous request to execute the job
+      Job request = jobControllerClient.submitJob(projectId, region, job);
+      String jobId = request.getReference().getJobId();
+      System.out.println(String.format("Submitted job \"%s\"", jobId));
+
+      // Wait for the job to finish
+      CompletableFuture<Job> finishedJobFuture =
+          CompletableFuture.supplyAsync(() ->
+           waitForJobCompletion(jobControllerClient, projectId, region, jobId));
+      int timeout = 10;
```

Comment:
nit: Can we inline timeout? It looks like it's only ever used once

Reply:
It's used twice, hence my reasoning for not wanting to inline it: once when calling `.get()`, the other in the `catch` block (line 152).